src/main/scala/roguelike/subsystems/WorkerSubSystem.scala

```diff
@@ -0,0 +1,73 @@
+package roguelike.subsystems
+
+import indigo.shared.IndigoLogger
+import indigo.shared.Outcome
+import indigo.shared.collections.Batch
+import indigo.shared.events.FrameTick
+import indigo.shared.events.GlobalEvent
+import indigo.shared.scenegraph.SceneUpdateFragment
+import indigo.shared.subsystems.SubSystem
+import indigo.shared.subsystems.SubSystemFrameContext
+import indigo.shared.subsystems.SubSystemId
+import org.scalajs.dom
+import org.scalajs.dom.Event
+import roguelike.workers.IndigoWorker
+
+import scala.annotation.nowarn.apply
+import scala.collection.mutable
+import scala.scalajs.js
+
+final case class WorkerSubSystem[A <: js.Any, B <: js.Any](
```

Comment:
This type looks a bit suspicious `A <: js.Any` - isn't that just anything?

Reply:
It's anything that can be converted to JS - this is so you can pass lists, scalars, objects etc. A bit messy, but the whole worker ecosystem is messy :-/